test-harness/index.ts

```diff
@@ -82,7 +83,19 @@ describe('cli acceptance tests', () => {
     test(scenario.test, async () => {
       const command = applyReplacements(scenario.command!, replacements);
       const { stderr, stdout, status } = await spawnNode(command, scenario.env, scenarioCwd);
-      replacements.date = String(new Date()); // this may introduce random failures, but hopefully they don't occur too often
+
+      // executing Date() before or after spawnNode were constantly leading to occasional mismatches,
+      // as the success of that approach was highly bound to the time spent on the actual spawnNode call
+      // this is a tad smarter, because instead of naively hoping the date will match, we try to extract the date from the actual output
+      // this regular expression matches "00:43:59" in "Thu Jul 08 2021 00:43:59 GMT+0200 (Central European Summer Time)"
+      const date = RegExp(escapeRegExp(String(Date())).replace(/(\d\d:){2}\d\d/, '(\\d\\d:){2}\\d\\d'));
+      Reflect.defineProperty(replacements, 'date', {
+        configurable: true,
+        enumerable: true,
+        get(): string {
+          return stdout.match(date)?.[0] ?? String(Date());
```

Comment:
Date without new returns a string, right? Then I guess we can just do:

```suggestion
          return stdout.match(date)?.[0] ?? Date();
```

or leave it how it was before:

```suggestion
          return stdout.match(date)?.[0] ?? String(new Date());
```

Reply:
True, you're right.